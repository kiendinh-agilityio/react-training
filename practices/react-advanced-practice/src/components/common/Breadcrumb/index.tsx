// Import radix ui
import { Box } from '@radix-ui/themes';

// Import common components
import { Text } from '@/components/common';

interface BreadcrumbProps {
  currentPage: string;
  label: string;
}

const Breadcrumb = ({ currentPage, label }: BreadcrumbProps) => (
  <Box>
    <Text className="flex gap-[4px] text-base">
      {label} /{' '}
      <Box as="span" className="text-xs text-dark dark:text-light">
        {currentPage}
      </Box>
    </Text>
  </Box>
);

export default Breadcrumb;
