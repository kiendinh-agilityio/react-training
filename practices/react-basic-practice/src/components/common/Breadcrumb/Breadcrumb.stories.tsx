import Breadcrumb from '.';

export default {
  component: Breadcrumb,
  title: 'Breadcrumb',
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
};

export const Tables = {
  args: {
    currentPage: 'Tables',
  },
};
