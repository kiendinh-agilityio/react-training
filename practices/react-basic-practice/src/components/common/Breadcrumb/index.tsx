interface BreadcrumbProps {
  currentPage: string;
}

const Breadcrumb = ({ currentPage }: BreadcrumbProps) => (
  <p className="text-base mb-1.5 text-xs">
    Pages / <span className="text-dark">{currentPage}</span>
  </p>
);

export default Breadcrumb;
