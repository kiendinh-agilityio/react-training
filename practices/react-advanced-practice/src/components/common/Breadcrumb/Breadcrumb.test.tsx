import { render } from '@testing-library/react';
import Breadcrumb from '.';

describe('Breadcrumb component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Breadcrumb currentPage="Tables" label="Pages" />);

    expect(container).toMatchSnapshot();
  });
});
