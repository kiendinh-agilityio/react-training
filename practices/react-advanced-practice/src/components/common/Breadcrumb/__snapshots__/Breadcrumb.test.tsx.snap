// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component matches the snapshot 1`] = `
<div>
  <div
    class="rt-Box flex items-center"
  >
    <p
      class="rt-Text rt-r-weight-regular text-sm text-base text-xs"
    >
      Pages
    </p>
    <span
      class="text-xs"
    >
      / 
      Tables
    </span>
  </div>
</div>
`;
